Feat/async reactor webhook url deprecation (#466)
Basis-Theory/developers.basistheory.com: docs/api/deprecations.mdx

| Application Key Regenerate                                          | April 17, 2024    | October 17, 2024  | Regenerating API Keys for an Application has been replaced with [Application Keys](/docs/api/applications/application-keys), to Migrate you will Create a new Application Key and remove the old key.                                                                                                                                                                                                           |
| Old Rate Limits                                                     | June 10, 2024     | July 16, 2024     | Rate limits have been deprecated in favor of [Rate Limits](/docs/api/rate-limits).                                                                                                                                                                                                                                                                                                                              |

> | Async Reactor `callback_url`                                        | November 1, 2024  | January 15, 2025  | Async Reactors will no longer support the `callback_url` attribute.                                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| Async Reactor `callback_url`                                        | November 1, 2024  | January 15, 2025  | Async Reactors will no longer support the `callback_url` attribute. For more information, see the section below on [Async Reactor callback_url Deprecation](#async-reactor-callback_url-deprecation).                                                                                                                                                                                                                                                                                                                                             |
```

You can continue to use your existing Reactors with `formula` until they are removed on a future date, but we advise you to migrate to `code` as soon as possible.
Keep in mind that after migrating to `code` the `formula` will be removed from the Reactor but can still be retrieved for consulting purposes until the shutdown date.

## Async Reactor `callback_url` Deprecation

Async Reactors will no longer support the `callback_url` attribute and will be replaced by the [Reactor Webhooks Events](/docs/concepts/what-are-reactors#asynchronous-reactors).

To migrate your Async Reactors, you will need to:

> 1. Create a new Webhook either with our [Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 

Reviewer: ```suggestion
1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api)/Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 
```

> 1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api)/Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 

Reviewer: ```suggestion
1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api)/[Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 
```

>     1. You may be able to use the existing URL for your rector processing

Reviewer: I don't understand this.

Author: trying to suggest they just re-use the existing url they use for `callback_url.. how is this?

```suggestion
    1. Set the webhook to the same url as you are sending in the `callback_url`property.
```

Reviewer: Great

> 1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api)/Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 
>     1. You may be able to use the existing URL for your rector processing

Reviewer 2: There may be a few typos in here:
```suggestion
1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api), [Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 
    1. Set the webhook to the same url as you are sending in the `callback_url` property.
```

Reviewer 2: You may also want to add another item 1. above that explains they need to update the url they are calling from `/react` to `/react-async`

2. Retrive the [Async Reactor Results](/docs/api/reactors#retrieve-result)

>     1. This ensures the system receiving the webhook has access to the results of the reactor.

Reviewer: ```suggestion
    - This ensures the system receiving the webhook has access to the results of the reactor.
```

Author: do you want a bullet here vs the `i`?

![image](https://github.com/user-attachments/assets/586d38be-0849-4a17-b789-425c89a0c777)

Reviewer 2: I'd also rephrase as 
```suggestion
    1. This allows the system receiving the webhook to access the results of the reactor.
```

3. Process them in the same way you are currently doing with the `callback_url`

> To cooridnate a rollout of these changes, we sugget you stop sending requests with `callback_url` when you roll out the new webhook handler code. This will avoid any overlap or duplicatation of processing.

Reviewer 2: ```suggestion
To coordinate a rollout of these changes, we suggest you stop sending requests with `callback_url` and change the request path from `/react` to `/react-async` when you roll out the new webhook handler code. This will avoid any overlap or duplication of processing.
```

You can continue to use your existing Reactors with `formula` until they are removed on a future date, but we advise you to migrate to `code` as soon as possible.
Keep in mind that after migrating to `code` the `formula` will be removed from the Reactor but can still be retrieved for consulting purposes until the shutdown date.

## Async Reactor `callback_url` Deprecation

> Async Reactors will no longer support the `callback_url` attribute and will be replaced by the [Reactor Webhooks Events](/docs/concepts/what-are-reactors#asynchronous-reactors).

Reviewer 2: This url points to the async reactors concept page, not the webhook events
```suggestion
[Async Reactors](/docs/concepts/what-are-reactors#asynchronous-reactors) will no longer support the `callback_url` attribute and will be replaced by the [Reactor Webhooks Events](/docs/api/webhooks/eventdata#reactors).
```

To migrate your Async Reactors, you will need to:

1. Create a new Webhook either with our [Management API](https://developers.basistheory.com/docs/api/webhooks/api)/[Terraform Provider](https://developers.basistheory.com/docs/sdks/server-side/terraform) or in the [Portal](https://portal.basistheory.com/settings/webhooks). 
    1. You may be able to use the existing URL for your rector processing

> 2. [Invoke a Reactor Asynchronous](/docs/api/reactors/#invoke-a-reactor-async) with the new `/reactors/:id/react-async` endpoint.

Reviewer 2: ```suggestion
2. [Invoke a Reactor Asynchronously](/docs/api/reactors/#invoke-a-reactor-async) with the new `/reactors/:id/react-async` endpoint.
```

> 3. Retrive the [Async Reactor Results](/docs/api/reactors#retrieve-result)

Reviewer: ```suggestion
3. Once your webhook is invoked, retrieve the [Async Reactor Results](/docs/api/reactors#retrieve-result)
```
